bundles/org.openhab.binding.coolmasternet/src/main/java/org/openhab/binding/coolmasternet/internal/ControllerHandler.java

```diff
@@ -44,116 +47,163 @@
-    private static final int SOCKET_TIMEOUT = 2000;
+public final class ControllerHandler extends BaseBridgeHandler {
+    private static final byte LF = "\n".getBytes(US_ASCII)[0];
+    private static final byte PROMPT = ">".getBytes(US_ASCII)[0];
+    private static final int LS_LINE_LENGTH = 36;
+    private static final int LS_LINE_TEMP_SCALE_OFFSET = 13;
+    private static final int MAX_VALID_LINE_LENGTH = LS_LINE_LENGTH * 20;
+    private static final int SINK_TIMEOUT_MS = 25;
+    private static final int SOCKET_TIMEOUT_MS = 2000;
 
+    private ControllerConfiguration cfg = new ControllerConfiguration();
+    private Unit<?> unit = SIUnits.CELSIUS;
     private final Logger logger = LoggerFactory.getLogger(ControllerHandler.class);
-    private final Object refreshLock = new Object();
     private final Object socketLock = new Object();
 
-    private @Nullable ScheduledFuture<?> refreshJob;
+    private @Nullable ScheduledFuture<?> poller;
     private @Nullable Socket socket;
 
-    public ControllerHandler(Bridge thing) {
+    public ControllerHandler(final Bridge thing) {
         super(thing);
     }
 
     @Override
     public void initialize() {
-        logger.debug("Initializing CoolMasterNet Controller handler...");
-        stopRefresh();
-        startRefresh();
+        cfg = getConfigAs(ControllerConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN);
+        determineTemperatureUnits();
+        stopPoller();
+        startPoller();
     }
 
     @Override
     public void dispose() {
-        stopRefresh();
-        logger.debug("Disconnecting CoolMasterNet Controller handler...");
+        updateStatus(ThingStatus.OFFLINE);
+        stopPoller();
         disconnect();
-        super.dispose();
     }
 
-    private void startRefresh() {
-        synchronized (refreshLock) {
-            ControllerConfiguration config = getConfigAs(ControllerConfiguration.class);
-            logger.debug("Scheduling new refresh job");
-            refreshJob = scheduler.scheduleWithFixedDelay(this::refreshHVACUnits, 0, config.refresh, TimeUnit.SECONDS);
+    /**
+     * Obtain the temperature unit configured for this controller.
+     *
+     * <p>
+     * CoolMasterNet defaults to Celsius, but allows a user to change the scale
+     * on a per-controller basis using the ASCII I/F "set deg" command. Given
+     * changing the unit is very rarely performed, there is no direct support
+     * for doing so within this binding.
+     *
+     * @return the unit as determined from the first line of the "ls" command
+     */
+    public Unit<?> getUnit() {
+        return unit;
+    }
+
+    private void determineTemperatureUnits() {
+        synchronized (socketLock) {
+            try {
+                checkConnection();
+                final String ls = sendCommand("ls");
+                if (ls.length() < LS_LINE_LENGTH) {
+                    throw new CoolMasterClientError("Invalid 'ls' response: '" + ls + "'");
```

Comment:
Why not use the constructor that handles formatting for you?

Reply:
I forgot I added one! :-) Fixed.